src/coreclr/jit/lowerxarch.cpp

```diff
@@ -4006,6 +4006,36 @@ GenTree* Lowering::TryLowerAndOpToAndNot(GenTreeOp* andNode)
     return andnNode;
 }
 
+void Lowering::LowerBswapOp(GenTreeOp* node)
+{
+    assert(node->OperIs(GT_BSWAP));
+
+    if (!comp->compOpportunisticallyDependsOn(InstructionSet_MOVBE))
+    {
+        return;
+    }
+
+    GenTree* operand = node->gtGetOp1();
+    if (operand->OperIs(GT_IND))
+    {
+        operand->SetContained();
```

Comment:
This has potential to unsafely move the indirection if there are interfering nodes between the indirection and `bswap`. You can check for interference using `IsSafeToContainMem`.